src/HttpHeader.cc

```diff
@@ -736,6 +736,37 @@ HttpHeader::refreshMask()
     }
 }
 
+bool
+HttpHeader::tooLarge() const
+{
+    size_t max = 0;
+    switch (owner) {
+    case hoRequest:
+        max = Config.maxRequestHeaderSize;
+        break;
+    case hoReply:
+        max = Config.maxReplyHeaderSize;
+        break;
+#if USE_HTCP
+    case hoHtcpReply:
+        // HTCP messages have a fixed header length
+        return false;
+#endif
+#if USE_OPENSSL
+    case hoErrorDetail:
+        // this case should not happen
```

Comment:
Why not? Let's give developers some clues, especially since you have already researched them. And if we are so sure that this code is unreachable, why are we not asserting instead of returning false here?

Reply:
Ok, since these error responses can be classified as other HTTP error responses, we can limit them with the same maxReplyHeaderSize.